src/main/scala/dpla/ingestion3/mappers/providers/DcMapping.scala

```diff
@@ -0,0 +1,138 @@
+package dpla.ingestion3.mappers.providers
+
+import java.net.URI
+
+import dpla.ingestion3.enrichments.normalizations.StringNormalizationUtils._
+import dpla.ingestion3.enrichments.normalizations.filters.{DigitalSurrogateBlockList, FormatTypeValuesBlockList}
+import dpla.ingestion3.mappers.utils.{Document, IdMinter, Mapping, XmlExtractor}
+import dpla.ingestion3.model.DplaMapData.{ExactlyOne, ZeroToOne}
+import dpla.ingestion3.model._
+import dpla.ingestion3.utils.Utils
+import org.json4s.JValue
+import org.json4s.JsonDSL._
+
+import scala.xml._
+
+
+class DcMapping extends Mapping[NodeSeq] with XmlExtractor with IdMinter[NodeSeq] {
+  // ID minting functions
+  override def useProviderName(): Boolean = false // TODO confirm prefix
+
+  override def getProviderName(): String = "dc" // TODO confirm prefix
+
+  override def getProviderId(implicit data: Document[NodeSeq]): String = // TODO confirm w/gretchen
+    extractString(data \ "header" \ "identifier")
+      .getOrElse(throw new RuntimeException(s"No ID for record $data")
+      )
+
+  // SourceResource mapping
+  override def collection(data: Document[NodeSeq]): Seq[DcmiTypeCollection] = // done
+    extractStrings(data \ "metadata" \\ "isPartOf")
+      .map(nameOnlyCollection)
+
+  override def contributor(data: Document[NodeSeq]): Seq[EdmAgent] = // done
+    extractStrings(data \ "metadata" \\ "contributor")
+      .map(nameOnlyAgent)
+
+  override def creator(data: Document[NodeSeq]): Seq[EdmAgent] = // done
+    extractStrings(data \ "metadata" \\ "creator")
+      .map(nameOnlyAgent)
+
+  override def date(data: Document[NodeSeq]): Seq[EdmTimeSpan] = // done
+    extractStrings(data \ "metadata" \\ "date")
+      .map(stringOnlyTimeSpan)
+
+  override def description(data: Document[NodeSeq]): Seq[String] =
+    extractStrings(data \ "metadata" \\ "description") ++
+      extractStrings(data \ "metadata" \\ "source")
+
+  override def format(data: Document[NodeSeq]): Seq[String] =
+    (extractStrings(data \ "metadata" \\ "format") ++
+      extractStrings(data \ "metadata" \\ "type"))
+      .map(_.applyBlockFilter(
+          DigitalSurrogateBlockList.termList ++
```

Comment:
I'd stick this concatenated list in a field somewhere rather than rebuilding it for every record.

Reply:
good tip